tools/crm_resource_print.c

```diff
@@ -235,75 +235,92 @@ cli_resource_print(pcmk__output_t *out, pe_resource_t *rsc,
     return pcmk_rc_ok;
 }
 
-// \return Standard Pacemaker return code
-int
-cli_resource_print_attribute(pcmk__output_t *out, pe_resource_t *rsc, const char *attr,
-                             const char *attr_set_type, pe_working_set_t * data_set)
-{
-    int rc = ENXIO;
-    unsigned int count = 0;
-    GHashTable *params = NULL;
-    const char *value = NULL;
-    pe_node_t *current = pe__find_active_on(rsc, &count, NULL);
-
-    if (count > 1) {
-        CMD_ERR("%s is active on more than one node,"
-                " returning the default value for %s", rsc->id, crm_str(attr));
-        current = NULL;
+static void
+add_resource_name(pcmk__output_t *out, pe_resource_t *rsc, bool is_xml) {
+    if (rsc->children == NULL) {
+        out->list_item(out, "resource", "%s", rsc->id);
+    } else {
+        for (GListPtr lpc = rsc->children; lpc != NULL; lpc = lpc->next) {
+            pe_resource_t *child = (pe_resource_t *) lpc->data;
+            add_resource_name(out, child, is_xml);
+        }
     }
+}
 
-    params = crm_str_table_new();
+PCMK__OUTPUT_ARGS("attribute", "pe_resource_t *", "const char *", "GHashTable *")
+static int
+attribute_default(pcmk__output_t *out, va_list args) {
+    pe_resource_t *rsc = va_arg(args, pe_resource_t *);
+    const char *attr = va_arg(args, const char *);
+    GHashTable *params = va_arg(args, GHashTable *);
 
-    if (pcmk__str_eq(attr_set_type, XML_TAG_ATTR_SETS, pcmk__str_casei)) {
-        get_rsc_attributes(params, rsc, current, data_set);
+    const char *value = g_hash_table_lookup(params, attr);
 
-    } else if (pcmk__str_eq(attr_set_type, XML_TAG_META_SETS, pcmk__str_casei)) {
-        /* No need to redirect to the parent */
-        get_meta_attributes(params, rsc, current, data_set);
+    if (value != NULL) {
+        out->begin_list(out, NULL, NULL, "Attributes");
+        out->list_item(out, attr, "%s", value);
+        out->end_list(out);
```

Comment:
I thought we might eventually display a list of attributes somewhere else, so it made sense to me to have attributes in a list everywhere, even if it were just a singleton list.  I suppose if I really thought we would want to display a list of attributes, I would do this outside of crm_resource, and it would take a list instead of a hash table, and I'd have to build a list in crm_resource.  If you think that's worth it, I can do it.

Otherwise, I can just change this.  I guess I'm not sure what the output would look like in that case.

Reply:
I think just the attribute value by itself, like the current output, is fine. It allows scripting like

    myparam=$(crm_resource -g myparam -r myrsc)

at least as long as the scripter isn't too worried about errors :)

Reply:
For text output, I'm already just printing it by itself - see attribute_text.  It's only HTML and XML that will get written out as a list of one item.